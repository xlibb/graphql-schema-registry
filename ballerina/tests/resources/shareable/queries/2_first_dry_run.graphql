query DryRun {
    dryRun(
        schema: {
            name: "subg2"
            url: "http://subg2"
            schema: """
            extend schema 
                @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@shareable"])
            
            type Query {
                name: String @shareable
                age: Int @shareable
                value: Boolean
            }
            """
        }
    ) {
        schema
        apiSchema
        version
        subgraphs {
            name
            schema
        }
        hints
        diffs {
            severity
            action
            subject
            location
            value
            fromValue
            toValue
        }
    }
}            