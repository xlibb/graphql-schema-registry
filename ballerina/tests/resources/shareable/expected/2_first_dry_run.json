{
    "data": {
        "dryRun": {
            "schema": "schema\n  @link(url: \"https://specs.apollo.dev/link/v1.0\")\n  @link(url: \"https://specs.apollo.dev/join/v0.3\", for: EXECUTION)\n{\n  query: Query\n}\n\ndirective @join__enumValue(graph: join__Graph!) repeatable on ENUM_VALUE\n\ndirective @join__field(graph: join__Graph, requires: join__FieldSet, provides: join__FieldSet, type: String, external: Boolean, override: String, usedOverridden: Boolean) repeatable on FIELD_DEFINITION | INPUT_FIELD_DEFINITION\n\ndirective @join__graph(name: String!, url: String!) on ENUM_VALUE\n\ndirective @join__implements(graph: join__Graph!, interface: String!) repeatable on OBJECT | INTERFACE\n\ndirective @join__type(graph: join__Graph!, key: join__FieldSet, extension: Boolean! = false, resolvable: Boolean! = true, isInterfaceObject: Boolean! = false) repeatable on OBJECT | INTERFACE | UNION | ENUM | INPUT_OBJECT | SCALAR\n\ndirective @join__unionMember(graph: join__Graph!, member: String!) repeatable on UNION\n\ndirective @link(url: String, as: String, for: link__Purpose, import: [link__Import]) repeatable on SCHEMA\n\nscalar join__FieldSet\n\nenum join__Graph {\n  SUBG2 @join__graph(name: \"subg2\", url: \"http://subg2\")\n}\n\nscalar link__Import\n\nenum link__Purpose {\n  \"\"\"\n  `SECURITY` features provide metadata necessary to securely resolve fields.\n  \"\"\"\n  SECURITY\n\n  \"\"\"\n  `EXECUTION` features provide metadata necessary for operation execution.\n  \"\"\"\n  EXECUTION\n}\n\ntype Query\n  @join__type(graph: SUBG2)\n{\n  name: String\n  age: Int\n  value: Boolean\n}",
            "apiSchema": "type Query {\n  name: String\n  age: Int\n  value: Boolean\n}",
            "version": "0.1.0",
            "subgraphs": [
                {
                    "name": "subg2",
                    "schema": "extend schema \n    @link(url: \"https://specs.apollo.dev/federation/v2.0\", import: [\"@shareable\"])\n\ntype Query {\n    name: String @shareable\n    age: Int @shareable\n    value: Boolean\n}"
                }
            ],
            "hints": [],
            "diffs": [
                {
                    "severity": "SAFE",
                    "action": "ADDED",
                    "subject": "FIELD",
                    "location": ["Query"],
                    "value": "name",
                    "fromValue": null,
                    "toValue": null
                },
                {
                    "severity": "SAFE",
                    "action": "ADDED",
                    "subject": "FIELD",
                    "location": ["Query"],
                    "value": "age",
                    "fromValue": null,
                    "toValue": null
                },
                {
                    "severity": "SAFE",
                    "action": "ADDED",
                    "subject": "FIELD",
                    "location": ["Query"],
                    "value": "value",
                    "fromValue": null,
                    "toValue": null
                }
            ]
        }
    }
}